docs/config/server-options.md

```diff
@@ -90,7 +90,7 @@ Configure custom proxy rules for the dev server. Expects an object of `{ key: op
 
 Note that if you are using non-relative [`base`](/config/shared-options.md#base), you must prefix each key with that `base`.
 
-Extends [`http-proxy`](https://github.com/http-party/node-http-proxy#options). Additional options are [here](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/server/middlewares/proxy.ts#L13). Note that [unlike http-proxy](https://github.com/http-party/node-http-proxy/issues/1669), the `changeOrigin` option will change both host and origin headers to match the target.
+Extends [`http-proxy`](https://github.com/http-party/node-http-proxy#options). Additional options are [here](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/server/middlewares/proxy.ts#L13). Note that [Unlike http-proxy](https://github.com/http-party/node-http-proxy/issues/1669), the `rewriteWsOrigin` option will rewrite a WebScocket request Origin header to match the target. **Exercise caution as rewriting the Origin can leave the proxying open to [CSRF attacks](https://owasp.org/www-community/attacks/csrf).**
```

Comment:
To keep it consistent with the other extended options, I think we should remove the sentence of `Note that [Unlike...` onwards till the end of the paragraph, and move further explanation of the feature in the jsdoc instead.

If we'd like additional documentation here, it could be in the example code block below that uses `rewriteWsOrigin`.

Reply:
https://github.com/vitejs/vite/pull/17563/commits/5503d2779b67e59f9cc60aa4c3b8d3eeebade931